server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -47,6 +45,33 @@ public static boolean isEmpty(Object[] array) {
         return array == null || array.length == 0;
     }
 
+    /**
+     * Eliminate duplicates from a list.
```

Comment:
nit: the `@param` javadocs already mention this but I think it's better to have it in the main javadoc too

```suggestion
     * Eliminate duplicates from a sorted list in-place.
```